python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -137,21 +138,33 @@ def run_iteration(
 
         evaluations_by_asset_key = {evaluation.asset_key: evaluation for evaluation in evaluations}
 
+        submit_job_inputs: List[Tuple[RunRequest, ExternalJob, ExternalExecutionPlan]] = []
+
         for run_request in run_requests:
             yield
+            submit_job_inputs.append(
+                get_asset_run_submit_input(
+                    run_request._replace(
+                        tags={
+                            **run_request.tags,
+                            ASSET_EVALUATION_ID_TAG: str(new_cursor.evaluation_id),
+                        }
+                    ),
+                    instance,
+                    workspace,
+                    asset_graph,
+                )
+            )
+
+        for submit_job_input in submit_job_inputs:
```

Comment:
I think we should leave a comment here laying out what we understand that if we eat an error here (or exit suddenly) between submitting runs and updating the cursor that we may re-submit duplicate runs

Reply:
could consider trying to put all these DB interactions including the cursor update as one transaction so it either all commits or none